kernel/src/engine/arrow_utils.rs

```diff
@@ -813,6 +813,22 @@ fn parse_json_impl(json_strings: &StringArray, schema: ArrowSchemaRef) -> DeltaR
     Ok(concat_batches(&schema, output.iter())?)
 }
 
+/// write an arrow RecordBatch to a JSON string by appending to a buffer.
+///
+/// TODO: this should stream data to the JSON writer and output an iterator.
+pub(crate) fn write_json(
+    data: Box<dyn Iterator<Item = Box<dyn EngineData>> + Send>,
```

Comment:
Out of curiosity, does this let you avoid the `Box`?
```suggestion
    data: impl Iterator<Item = Box<dyn EngineData>> + Send,
```

Reply:
haven't found a way around it yet..

Reply:
jk it worked lol